### PR TITLE
Audin fix

### PR DESCRIPTION
**Pull request by akallabeth**
* Fixes a memory corruption in audin argument parser
* Move to preallocated dynamic logger

### CODE REVIEW

File: channels/audin/client/audin_main.c [hunk 1]
```diff
@@ -327,50 +339,52 @@ static UINT audin_receive_wave_data(const BYTE* data, int size, void* user_data)
...
+static UINT audin_process_open(AUDIN_PLUGIN* audin, AUDIN_CHANNEL_CALLBACK* callback, wStream* s)
 {
-	AUDIN_CHANNEL_CALLBACK* callback = (AUDIN_CHANNEL_CALLBACK*) pChannelCallback;
-	AUDIN_PLUGIN* audin = (AUDIN_PLUGIN*) callback->plugin;
 	audinFormat* format;
 	UINT32 initialFormat;
 	UINT32 FramesPerPacket;
 	UINT error = CHANNEL_RC_OK;
+
+	if (!audin || !callback || !s)
```
Comment (hardening) [inline channels/audin/client/audin_main.c]: You can drop these checks, the caller is already doing them.

File: channels/audin/client/audin_main.c [hunk 2]
```diff
@@ -412,20 +426,26 @@ static UINT audin_process_open(IWTSVirtualChannelCallback* pChannelCallback, wSt
...
 {
-	AUDIN_CHANNEL_CALLBACK* callback = (AUDIN_CHANNEL_CALLBACK*) pChannelCallback;
-	AUDIN_PLUGIN* audin = (AUDIN_PLUGIN*) callback->plugin;
 	UINT32 NewFormat;
 	audinFormat* format;
 	UINT error = CHANNEL_RC_OK;
+
+	if (!audin || !callback || !s)
+		return ERROR_INTERNAL_ERROR;
+
```
Comment (hardening) [inline channels/audin/client/audin_main.c]: The same here, caller is doing all the necessary checks

**Comment by akallabeth** [comment]
superseeded by #4453

**Comment by bmiklautz** [comment]
When testing with pulse I get the following:
```
xfreerdp /sound:sys:pulse /microphone:sys:pulse /v:win10
[ERROR][com.freerdp.channels.audin.client] - Close failed with errorcode 87
```
Sound doesn't work (works with current master).

**Comment by akallabeth** [comment]
@bmiklautz Good catch, fixed the issue.

**Pull request merged**